File: frontend-v1/types/owner.ts

// =======================
// Owner Request DTO
// =======================
export interface OwnerRequest {
  firstName: string;   // 0 - 100 characters
  lastName: string;    // 0 - 100 characters
  phone: string;       // 0 - 20 characters
  email: string;       // email format, 0 - 100 characters
  address: string;     // 0 - 255 characters
}

// =======================
// Owner Response DTO
// =======================
export interface OwnerResponse {
  ownerId: number;     // int64 -> number
  firstName: string;
  lastName: string;
  phone: string;
  email: string;
  address: string;
}

File: backend/src/main/java/com/backend/model/Owner.java
package com.backend.model;


import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Table(name = "owners")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Owner {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "owner_id")
    private Long ownerId;

    @Column(name = "first_name", nullable = false, length = 100)
    private String firstName;

    @Column(name = "last_name", nullable = true, length = 100)
    private String lastName;

    @Column(name = "phone", nullable = false, unique = true, length = 20)
    private String phone;

    @Column(name = "email", nullable = false, unique = true, length = 100)
    private String email;

    @OneToMany(mappedBy = "owner", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Patient> patients;

    @OneToMany(mappedBy = "owner", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Billing> billings;

    @Column(name = "address", nullable = true, length = 255)
    private String address;
}
